client/web/src/enterprise/campaigns/preview/list/PreviewAction.tsx

```diff
@@ -0,0 +1,130 @@
+    }
+    if (node.operations.length === 0) {
+        if (node.targets.__typename === 'VisibleApplyPreviewTargetsDetach') {
+            return <PreviewActionDetach className={className} />
+        }
+        return <PreviewActionNoAction className={className} />
+    }
+    if (node.operations.includes(ChangesetSpecOperation.IMPORT)) {
+        return <PreviewActionImport className={className} />
+    }
+    if (node.operations.includes(ChangesetSpecOperation.PUBLISH)) {
+        return <PreviewActionPublish className={className} />
+    }
+    if (node.operations.includes(ChangesetSpecOperation.PUBLISH_DRAFT)) {
+        return <PreviewActionPublishDraft className={className} />
+    }
+    if (node.operations.includes(ChangesetSpecOperation.CLOSE)) {
+        return <PreviewActionClose className={className} />
+    }
+    if (node.operations.includes(ChangesetSpecOperation.REOPEN)) {
+        return <PreviewActionReopen className={className} />
+    }
+    if (node.operations.includes(ChangesetSpecOperation.UNDRAFT)) {
+        return <PreviewActionUndraft className={className} />
+    }
+    if (
+        node.operations.includes(ChangesetSpecOperation.UPDATE) ||
+        node.operations.includes(ChangesetSpecOperation.PUSH)
+    ) {
+        return <PreviewActionUpdate className={className} />
+    }
+    return <PreviewActionUnknown operations={node.operations.join(' => ')} className={className} />
+}
+
+const iconClassNames = 'm-0 text-nowrap d-block d-sm-flex flex-column align-items-center justify-content-center'
+
+export const PreviewActionPublish: React.FunctionComponent<{ className?: string }> = ({ className }) => (
+    <div className={classNames(className, iconClassNames)}>
+        <UploadIcon data-tooltip="This changeset will be published to its code host" />
+        <span>Publish</span>
+    </div>
+)
+export const PreviewActionPublishDraft: React.FunctionComponent<{ className?: string }> = ({ className }) => (
+    <div className={classNames(className, iconClassNames)}>
+        <UploadIcon className="text-muted" data-tooltip="This changeset will be published as draft to its code host" />
+        <span>Publish draft</span>
+    </div>
+)
+export const PreviewActionImport: React.FunctionComponent<{ className?: string }> = ({ className }) => (
+    <div className={classNames(className, iconClassNames)}>
+        <ImportIcon data-tooltip="This changeset will be imported and tracked in this campaign" />
+        <span>Import</span>
+    </div>
+)
+// TODO: This is currently correct, but as soon as we have a detach reconciler operation, that should be taken into account.
+export const PreviewActionClose: React.FunctionComponent<{ className?: string }> = ({ className }) => (
+    <div className={classNames(className, iconClassNames)}>
+        <CloseCircleOutlineIcon className="text-danger" data-tooltip="This changeset will be closed on the codehost" />
```

Comment:
We have some inconsistency between `codehost` and `code host`; I've arbitrarily picked `code host` as the right version and will suggest accordingly in this file, but I'm not really that fussed as long as they're the same.

```suggestion
        <CloseCircleOutlineIcon className="text-danger" data-tooltip="This changeset will be closed on the code host" />
```